Cancel interaction documentation - Fall 2019 documentation (#156)
smartdevicelink/sdl_app_library_guides: docs/Displaying a User Interface/Slider/index.md

##### Objective-C
```objc
// Create a slider with number of ticks, starting position 'tick number', a header message, an optional footer message, and a timeout of 30 seconds

> SDLSlider *sdlSlider = [[SDLSlider alloc] initWithNumTicks:5 position:1 sliderHeader:@"This is a Header" sliderFooter:@"Static Footer" timeout:30000 cancelID:45];

Reviewer: These should have sections to add in data like the Alert section

##### Swift
```swift
// Create a slider with number of ticks, starting position 'tick number', a header message, an optional footer message, and a timeout of 30 seconds

> let slider =  SDLSlider(numTicks: 5, position: 1, sliderHeader: "This is a Header", sliderFooter: "Static Footer", timeout: 30000, cancelID: 45)

Reviewer: See above

```
##### Swift
```swift
// Max length 500 chars
sdlSlider.sliderFooter = ["Static Footer"]
```
!@

@![android,javaSE,javaEE]
`TODO: add info`
!@

### Dynamic Footer
This type of footer will have a different message displayed for each position of the slider. The footer is an optional paramater. The footer message displayed will be based off of the slider's current position. The footer array should be the same length as `numTicks` because each footer must correspond to a tick value. Or, you can pass @![iOS]`nil`!@@![android,javaSE,javaEE]`null`!@ to have no footer at all.

@![iOS]
##### Objective-C
```objc
// Array length 1 - 26, Max length 500 chars
NSArray<NSString *> *footers = @[@"Footer 1", @"Footer 2", @"Footer 3"];
sdlSlider.sliderFooter = footers;
```
##### Swift
```swift
// Array length 1 - 26, Max length 500 chars
let footers = ["Footer 1", "Footer 2", "Footer 3"]
sdlSlider.sliderFooter = footers
```
!@

@![android,javaSE,javaEE]
`TODO: add info`
!@

### Cancel ID
An ID for this specific slider to allow cancellation through the `CancelInteraction` RPC.
@![iOS]
##### Objective-C
```objc
sdlSlider.cancelID = @(45);
```
##### Swift
```swift

> sdlSlider.sliderFooter = 45

Reviewer: ```suggestion
sdlSlider.cancelID = 45
```

NSArray<NSString *> *footers = @[@"Footer 1", @"Footer 2", @"Footer 3"];

// Create a slider with number of ticks, starting position 'tick number', a header message, and an optional footer array, and a timeout of 30 seconds

> SDLSlider *sdlSlider = [[SDLSlider alloc] initWithNumTicks:3 position:1 sliderHeader:@"This is a Header" sliderFooters:footers timeout:30000 cancelID:45];

Reviewer: See above

let footers = ["Footer 1", "Footer 2", "Footer 3"]

// Create a slider with number of ticks, starting position 'tick number', a header message, and an optional footer array, and a timeout of 30 seconds

> let slider =  SDLSlider(numTicks: 3, position: 1, sliderHeader: "This is a Header", sliderFooters: footers, timeout: 30000, cancelID: 45)

Reviewer: See above

sdlManager.sendRPC(slider);
```
!@

> ## Canceling a Specific Slider

Reviewer: ```suggestion
## Dismissing a Slider (RPC 6.0+)
```

> If you are connected to a head unit with SDL Core v6.0+, you can dismiss a displayed slider before the timeout has elapsed.

Reviewer: ```suggestion
You can dismiss a displayed slider before the timeout has elapsed by dismissing either a specific slider or the current slider.
```

> If connected to older head units that do not support this feature, the cancel request will be ignored, and the slider will persist on the screen until the timeout has elapsed or the user dismisses by selecting a position or canceling.

Reviewer: Make into a note

Author: done

### Dismissing a Specific Slider

@![iOS]
##### Objective-C
```objc
// Assign a unique cancel id to the slider
UInt32 cancelID = 45;

> slider.cancelID = @(cancelID);

Reviewer: See above

> // Use the cancel id to dismiss the slider

Reviewer: ```suggestion
// `cancelID` is the ID that you assigned when creating and sending the alert
```

SDLCancelInteraction *cancelInteraction = [[SDLCancelInteraction alloc] initWithSliderCancelID:cancelID];
[self.sdlManager sendRequest:cancelInteraction withResponseHandler:^(__kindof SDLRPCRequest * _Nullable request, __kindof SDLRPCResponse * _Nullable response, NSError * _Nullable error) {
    if (![response.resultCode isEqualToEnum:SDLResultSuccess]) { return; }
    <#The slider was canceled successfully#>
}];
```

##### Swift
```swift
// Assign a unique cancel id to the slider
let cancelID: UInt32 = 45

> slider.cancelID = cancelID as NSNumber

Reviewer: See above

> // Use the cancel id to dismiss the slider

Reviewer: ```suggestion
// `cancelID` is the ID that you assigned when creating and sending the alert
```

let cancelInteraction = SDLCancelInteraction(sliderCancelID: cancelID)
sdlManager.send(request: cancelInteraction) { (request, response, error) in
    guard response?.resultCode == .success else { return }
    <#The slider was canceled successfully#>
}
```
!@

@![android,javaSE,javaEE]
```java
// Assign a unique cancel id to the slider
final Integer cancelID = 45;

> slider.setCancelID(cancelID);

Reviewer: See above

> // Use the cancel id to dismiss the slider

Reviewer: ```suggestion
// `cancelID` is the ID that you assigned when creating and sending the alert
```

CancelInteraction cancelInteraction = new CancelInteraction(FunctionID.SLIDER.getId(), cancelID);
cancelInteraction.setOnRPCResponseListener(new OnRPCResponseListener() {
    @Override
    public void onResponse(int correlationId, RPCResponse response) {
        if (response.getSuccess()){
            Log.i(TAG, "Slider was dismissed successfully");
        }
    }

    @Override
    public void onError(int correlationId, Result resultCode, String info) {
        Log.e(TAG, "onError: "+ resultCode+ " | Info: "+ info );
    }
});
sdlManager.sendRPC(cancelInteraction);
```
!@

> ### Dismissing Current Slider

Reviewer: ```suggestion
### Dismissing the Current Slider
```
